### PR TITLE
Add output-check flag option

### LINKED ISSUE #456: Return name of file when using file output
### What problem are you facing?

Using GitLab CI to check if terraform docs are up to date. There is no way to know if running `terraform-docs` updated a file when run. 

Using `terraform fmt` as an example, we can easily check if a file changed and fail the pipeline if true. 

### How could terraform-docs help solve your problem?

When a file changed, returning the filename would allow us to know if a file changed and fail if true. Meaning the docs were not updated when the module changed.

### PR DESCRIPTION
**Pull request by rickhlx**
### Description of your changes

- Add a `--output-check` flag that only works when outputting to a file.
   - Exit code 0 when no changes
   - Exit code 1 when changes found

Also: 
- Return filename when output file updated (or differences found)

Fixes #456 

I have:

- [x] Read and followed terraform-docs' [contribution process].
- [x] All tests pass when I run `make test`.

### How has this code been tested

[contribution process]: https://git.io/JtEzg

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -66,6 +66,7 @@ func NewCommand() *cobra.Command {
 	cmd.PersistentFlags().StringVar(&config.Output.File, "output-file", "", "File in module directory to insert output into (default \"\")")
 	cmd.PersistentFlags().StringVar(&config.Output.Mode, "output-mode", "inject", "Output to file method ["+cli.OutputModes+"]")
 	cmd.PersistentFlags().StringVar(&config.Output.Template, "output-template", cli.OutputTemplate, "Output template")
+	cmd.PersistentFlags().BoolVar(&config.Output.Check, "output-check", false, "check if output is generated doc (default false)")
```
Comment (khos2ow) [inline cmd/root.go:69]: I think this is bit vague. Maybe something like:

```text
Check if content of output file is up to date
```

**Review by khos2ow — COMMENTED** [review]
Sorry for the delay, can you please check my comments below and also rebase on master?

File: cmd/root.go (hunk 1 above)
Comment (rickhlx) [reply cmd/root.go]: Sounds good to me

**Comment by rickhlx** [comment]
@khos2ow implemented the changes suggested, thanks for reviewing!

File: internal/cli/config.go [hunk 2]
```diff
@@ -187,6 +187,8 @@ const (
 	outputBeginComment = "<!-- BEGIN_TF_DOCS -->"
 	outputContent      = "{{ .Content }}"
 	outputEndComment   = "<!-- END_TF_DOCS -->"
+
+	outputCheck = false
```
Comment (khos2ow) [inline internal/cli/config.go:191]: This too should be removed. Make sure to run `make lint` locally, or check the [output](https://github.com/terraform-docs/terraform-docs/pull/458/checks?check_run_id=2504783389) of GitHub Action.

File: .editorconfig [hunk 3]
```diff
@@ -18,6 +18,7 @@ indent_size = 4
 
 [*.md]
 trim_trailing_whitespace = true
+indent_size = 2
```
Comment (khos2ow) [inline .editorconfig:21]: To be fair, this is very opinionated. Please revert it back.

**Comment by rickhlx** [comment]
Thanks for the comments @khos2ow, fixed.

File: internal/cli/writer.go [hunk 4]
```diff
@@ -157,5 +159,26 @@ func (fw *fileWriter) write(filename string, p []byte) (int, error) {
 	if fw.writer != nil {
```
Comment (khos2ow) [inline internal/cli/writer.go:159]: I suggest converting the whole function as follow:

```go
func (fw *fileWriter) write(filename string, p []byte) (int, error) {
	// if run in check mode return exit 1
	if fw.check {
		f, err := os.ReadFile(filename)
		if err != nil {
			return 0, err
		}

		// check for changes and print changed file
		if !bytes.Equal(f, p) {
			return 0, fmt.Errorf("%s is out of date", filename)
		}

		fmt.Printf("%s is up to date.\n", filename)
		return 0, nil
	}

	if fw.writer != nil {
		return fw.writer.Write(p)
	}

	fmt.Printf("%s updated successfully.\n", filename)
	return len(p), os.WriteFile(filename, p, 0644)
}
```

Comment (khos2ow) [reply internal/cli/writer.go:159]: Consecutively this test case can also be added in `writer_test.go` at the end of `TestFileWriter`:

```go
		"ModeReplaceOutOfDate": {
			file:     "mode-replace.md",
			mode:     "replace",
			check:    true,
			template: OutputTemplate,
			begin:    outputBeginComment,
			end:      outputEndComment,
			writer:   nil,

			expected: "",
			wantErr:  true,
			errMsg:   "testdata/writer/mode-replace.md is out of date",
		},
```

Note that you also need to add `check bool` to the struct and `check: false,` to all the other test cases.

Comment (khos2ow) [reply internal/cli/writer.go:159]: @rickhlx did you want to finish off your changes here? or would you rather me adding these new changes instead? Either way is great!

Comment (rickhlx) [reply internal/cli/writer.go:159]: Sorry! Been running around a bit. I'll take care of these changes, no problem. Thanks for the comments!

**Comment by khos2ow** [comment]
@rickhlx I accidentally pushed to the wrong branch and closed this PR! I apologize. I'm gonna open another PR with your changes.

**Pull request closed without merging**